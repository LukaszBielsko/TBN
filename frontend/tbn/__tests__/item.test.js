import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ItemComponent from '../components/Item';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const fakeItem = {
  _id: 'abc',
  title: 'product',
  place: 'china',
  description: 'great product',
  largeImage: 'large.jpg',
};

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent {...fakeItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  // let wrapper;
  // beforeEach(() => {
  //   wrapper = shallow(<ItemComponent {...fakeItem} />);
  // });
  // it('renders component properly', () => {
  //   expect(wrapper.length).toBe(1);
  // });
  // it('displays proper place', () => {
  //   const place = wrapper.find('.item-place');
  //   expect(place.text()).toBe('china');
  // });
  // it('displays proper description', () => {
  //   const description = wrapper.find('.item-description');
  //   expect(description.text()).toBe('great product');
  // });
  // it('proper img src is passed', () => {
  //   const img = wrapper.find('img');
  //   expect(img.prop('src')).toEqual('large.jpg');
  // });
  // it('proper id is passed', () => {
  //   /* TODO check for id */
  // });
});
