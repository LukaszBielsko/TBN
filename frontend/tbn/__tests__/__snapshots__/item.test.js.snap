// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<Item__Card>
  <Link
    href={
      Object {
        "pathname": "item",
        "query": Object {
          "id": "abc",
        },
      }
    }
  >
    <img
      className="image"
      src="large.jpg"
    />
  </Link>
  <Link
    href={
      Object {
        "pathname": "item",
        "query": Object {
          "id": "abc",
        },
      }
    }
  >
    <div
      className="item-header"
    >
      <div
        className="item-place"
      >
        china
      </div>
      <div
        className="item-title"
      >
        product
      </div>
      <div
        className="item-description"
      >
        great product
      </div>
    </div>
  </Link>
  <div
    className="item-buttons"
  >
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <DelteItem
      id="abc"
    >
       Delete this item 
    </DelteItem>
    <button>
       Button
    </button>
  </div>
</Item__Card>
`;
